### PR TITLE
Adding dockerfile and ci/cd

### LINKED ISSUE #869: Any objection if I add Dockerfile here ?
wiki page https://github.com/stefanberger/swtpm/wiki/Dockerfiles is not enough...

something like this is super easy and helpful:

```
$ cat Dockerfile
FROM debian:12.6-slim

RUN apt-get update && \
    apt-get install --no-install-recommends -y swtpm=0.7.1-1.3 && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

ENTRYPOINT ["/usr/bin/swtpm"]
```
lint passed:
```
$ docker run --rm -i ghcr.io/hadolint/hadolint < Dockerfile
$
```

build:
```
$ docker build -t swtpm:0.7.1 - < Dockerfile
Sending build context to Docker daemon  2.048kB
Step 1/3 : FROM debian:12.6-slim
 ---> 3eeb141ea1a1
Step 2/3 : RUN apt-get update &&     apt-get install --no-install-recommends -y swtpm=0.7.1-1.3 &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> a9e31f6fdb72
Step 3/3 : ENTRYPOINT ["/usr/bin/swtpm"]
 ---> Using cache
 ---> 6fb7ab4b45f6
Successfully built 6fb7ab4b45f6
Successfully tagged swtpm:0.7.1
```
run:
```
$ docker run --rm -it swtpm:0.7.1 --version
TPM emulator version 0.7.1, Copyright (c) 2014-2021 IBM Corp.
```

### PR DESCRIPTION
**Pull request by glimchb**
Fixes #869

- Adding alpine based dockerfile
- ci: adding linter and builder github action
- ci: add testing for dockerfile
- ci: add new release action to run when published

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Release
+
+on:
+    workflow_dispatch:
+    release:
+        types: [published]
+
```
Comment (stefanberger) [inline .github/workflows/release.yaml:7]: So this will only push once v0.10 is release?

Comment (glimchb) [reply .github/workflows/release.yaml:7]: `.github/workflows/docker.yaml` will push after afery PR merged
`.github/workflows/release.yaml` will push on every release with a tag

**Review by stefanberger — APPROVED** [review]

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      # Workaround: https://github.com/docker/build-push-action/issues/461
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v3.0.0
+
+    #   - name: Log in to Docker Hub
+    #     if: github.event_name != 'pull_request'
+    #     uses: docker/login-action@v3.0.0
+    #     with:
+    #       username: ${{ secrets.DOCKERHUB_USERNAME }}
+    #       password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (stefanberger) [inline .github/workflows/docker.yaml:44]: I now set these two as repository secrets. So theoretically this should work.

Comment (stefanberger) [reply .github/workflows/docker.yaml:44]: I am not sure how to enable this file here and see it running already as part of the PR...

Comment (glimchb) [reply .github/workflows/docker.yaml:44]: I will enable this in next patch, no problem, once you added secrets, should be easy for me to add this

**Comment by glimchb** [comment]
@stefanberger can you merge this , please ? Travis CI passed

**Comment by stefanberger** [comment]
Thanks!

File: .github/workflows/docker.yaml (hunk 2 above)
Comment (glimchb) [reply .github/workflows/docker.yaml:44]: opened #876

**Comment by stefanberger** [comment]
I had to fix the Dockerfile for github ci/cd since it started to fail due to an update of libtpms adding a new API call that then wasn't available to swtpm because the libtpms build didn't happen and the old version of libtpms was used: PR #898  
There were lots of broken github builds for the build action lately. I wouldn't mind it skipping the libtpms build for as long as it hasn't been updated but when an update happens it should build again. I suppose there's no way of doing this easily...

**Comment by glimchb** [comment]
@stefanberger one way to do it easily is add `libtpms` as a git submodule of `swtpm`
```
cd swtpm
git submodule add ../libtpms
```
in this way every time to checkout `swtpm` it will bring correct `libtpms` with it

so docker build and all other builds, will just do normal checkout of `swtpm` and don;t need to worry about `libtpm`

every time you need to bump the `libtpm`, you just checkout and commit as usual...

WDYT ?


another option - I exposed `LIBTPMS_BRANCH` argument, so we can just pass specific commit/tag instead of master...

another option - we use `--no-cache` in docker build... so it will always use latest master...

**Pull request merged**